web-api/src/embedding.rs

```diff
@@ -17,14 +17,26 @@ use xayn_ai_bert::{AvgEmbedder, Config as EmbedderConfig, NormalizedEmbedding};
 
 use crate::{app::SetupError, error::common::InternalError, utils::RelativePathBuf};
 
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(tag = "type", rename_all = "lowercase", deny_unknown_fields)]
+pub enum Config {
+    Pipeline(Pipeline),
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self::Pipeline(Pipeline::default())
+    }
+}
+
 #[derive(Debug, Deserialize, Serialize)]
 #[serde(default, deny_unknown_fields)]
-pub struct Config {
-    pub(crate) directory: RelativePathBuf,
+pub struct Pipeline {
+    pub(crate) directory: String,
```

Comment:
:wrench: we should keep the `RelativePathBuf` here. this just requires to remove the conversion in l.58 and add it in l.98

```suggestion
    pub(crate) directory: RelativePathBuf,
```